http/file_server_test.ts

```diff
@@ -991,3 +991,94 @@ Deno.test(
     assertStringIncludes(await res.text(), "Hello World");
   },
 );
+
+Deno.test(
+  "file_server returns 304 for requests with if-none-match set with the etag but with W/ prefixed etag in request headers.",
+  async () => {
+    await startFileServer();
+    try {
+      const testurl = "http://localhost:4507/testdata/desktop.ini";
+      const fileurl = new URL("./testdata/desktop.ini", import.meta.url);
+      let etag: string | undefined | null;
+      let last_modified: string | null | undefined;
+      {
+        const res = await fetch(
+          testurl,
+          {
+            headers: [
```

Comment:
Most headers look irrelevant to the purpose of the test case. I guess we need only `Accept-Encoding` to get `W/` prefixed etag.